src/content/docs/best-practices/typescript.mdx

```diff
@@ -200,7 +200,7 @@ function withDelay<
   Params extends PathParams,
   RequestBodyType extends DefaultBodyType,
   ResponseBodyType extends DefaultBodyType
->(durationMs: number, resolver: HttpResponseResolver): HttpResponseResolver {
+>(durationMs: number, resolver: HttpResponseResolver<Params, RequestBodyType, ResponseBodyType>): HttpResponseResolver<Params, RequestBodyType, ResponseBodyType> {
```

Comment:
A nice find!

Our automated test for this looks precisely like you are suggesting: https://github.com/mswjs/msw/blob/16fadfeb872c156cc610977cc5507630d59146ad/test/typings/custom-resolver.test-d.ts#L20-L34